pkg/table/table.go

```diff
@@ -251,10 +274,31 @@ type IndexRowLayoutOption []int
 // It's mapping from index ID to the layout of the index.
 type IndexesLayout map[int64]IndexRowLayoutOption
 
+// GetIndexLayout returns the layout of the specified index.
+func (idx IndexesLayout) GetIndexLayout(idxID int64) IndexRowLayoutOption {
+	if idx == nil {
+		return nil
+	}
+	return idx[idxID]
+}
+
 func (idx IndexesLayout) applyRemoveRecordOpt(opt *RemoveRecordOpt) {
 	opt.indexesLayoutOffset = idx
 }
 
+// ColumnSizeOption records the column size information.
+// If the column is not pruned, we can calculate its acurate size by the real data.
+// Otherwise, we use the estimated avg size given by table statistics and field type information.
+type ColumnSizeOption struct {
+	NotPruned        *bitset.BitSet
+	AvgSizes         []float64
+	PublicColsLayout []int
+}
```

Comment:
You didn't mention why do we need this and what's this for.
There are also no comments for each field, which is not easy to understand.